### DISCUSSION TITLE
multiple routes/paths for one function

### DISCUSSION THREAD

**Discussion by pozytanie** [fastapi/fastapi#4299] [Questions] [answered]
Hi,
is it possible to have multiple routes/paths for one function? 

For example:

```
@router.post(
    "/files",
    status_code=http_status.HTTP_201_CREATED,
    summary="Upload file",
    deprecated=True
)
@router.post(
    "/files/binary",
    status_code=http_status.HTTP_201_CREATED,
    summary="Upload file",
)
async def upload_file():```

is it ok or should I use something else?

**Comment by shashankrnr32** [comment, marked as answer]
Adding multiple decorators with different API endpoints(with different endpoint configuration, like deprecated) should work out of the box as you expected. The above example you provided should work fine with one endpoint showing up as deprecated in documentation.

**Note:** You can also use `router.add_api_route(endpoint, func, **kwargs)` (with `methods=["POST", ]` in the kwargs) and add the same config without using decorators.

**Reply by eykamp** [reply]
I can confirm that the multiple decorators approach worked for me (using GETs), and I think it is a very clean solution.
